src/api.rs

```diff
@@ -136,18 +138,29 @@ pub unsafe extern "C" fn verify_piece_inclusion_proof(
 
     info!("verify_piece_inclusion_proof: {}", "start");
 
-    let bytes = from_raw_parts(piece_inclusion_proof_ptr, piece_inclusion_proof_len);
+    let bytes = Ok(()).and_then(|_| {
+        ensure!(
+            !piece_inclusion_proof_ptr.is_null(),
+            "piece_inclusion_proof_ptr must not be null"
+        );
+        Ok(from_raw_parts(
```

Comment:
Idea: What if you added a new function to rust-fil-ffi-toolkit which was called `from_raw_parts_safe` which was of type `pub unsafe fn from_raw_parts_safe<'a, T, E>(data: *const T, len: usize) -> Result<&'a [T], E>`? It would allow you to remove some of the null checking.

Reply:
one day..